### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by joaanna**
Adding test that save different graphs, load them and ensure the variable values are as expected.

### CODE REVIEW

**Comment by igordertigor** [comment]
Kannst Du mal rebase master machen?

File: tests/test_tfpb.py [hunk 1]
```diff
@@ -105,5 +126,89 @@ def test_clear_devices_removes_all_device_specs(self):
                               for node in clear_graph_def.node]))
 
 
+class TestVariablesAfterSaving(tf.test.TestCase):
```
Comment (igordertigor) [inline tests/test_tfpb.py]: The tests in this class are all very similar and the only thing that differs seems to be the function used to build the graph and the expected value of the variable. You could move the repeated code into a function of the following form
```python
def save_graph_reload_and_check(graph_builder, expected_output):
    "Here goes the code to run the test"
```
That function could be called repeatedly.

Comment (joaanna) [inline tests/test_tfpb.py]: ok

Comment (joaanna) [inline tests/test_tfpb.py]: ok

File: tests/test_tfpb.py [hunk 2]
```diff
@@ -105,5 +126,73 @@ def test_clear_devices_removes_all_device_specs(self):
...
+                                                          10.))
+
+    def test_variable_graph(self):
+        self.assertEqual(True,
+                         self.save_graph_reload_and_check(self.build_variable_graph,
+                                                          1.))
+
+    def test_trained_graph(self):
+        self.assertNotEqual(True,
+                            self.save_graph_reload_and_check(self.build_variable_graph,
```
Comment (igordertigor) [inline tests/test_tfpb.py]: build_training_graph

File: tests/test_tfpb.py [hunk 3]
```diff
@@ -105,5 +126,73 @@ def test_clear_devices_removes_all_device_specs(self):
...
+                                                             10.))
```
Comment (igordertigor) [inline tests/test_tfpb.py]: not equal 1

File: tests/test_tfpb.py [hunk 4]
```diff
@@ -105,5 +126,73 @@ def test_clear_devices_removes_all_device_specs(self):
...
+                z = tf.placeholder("float", name='Target')
+                w = tf.Variable(np.array([1.]), name='Factor', dtype=tf.float32)
+                y = tf.multiply(w, x, name='Output')
+                cost = tf.pow(y - z, 2)
+                optimizer = tf.train.GradientDescentOptimizer(0.1).minimize(cost)
+                self.session.run(tf.global_variables_initializer())
+                for i in range(10):
+                    self.session.run(optimizer, feed_dict={x: 1, z: 10})
+
+    def save_graph_reload_and_check(self, graph_builder, expected_output):
```
Comment (igordertigor) [inline tests/test_tfpb.py]: Don't take expected_output, but return loaded value and compare in test. (Rename function to `save_graph_and_reload`

File: tests/test_tfpb.py [hunk 5]
```diff
@@ -105,5 +126,73 @@ def test_clear_devices_removes_all_device_specs(self):
...
+        with self.session.as_default():
+            graph_builder()
+            tfpb.save_protobuf(self.session.graph.as_graph_def(),
+                               self.fname,
+                               output_nodes=('Output:0',),
+                               freeze=True,
+                               session=self.session)
+            node = tfpb.load_protobuf(self.fname,
```
Comment (igordertigor) [inline tests/test_tfpb.py]: Here you need to reset the graph.

**Pull request merged**